File: javaCode/SmithGame/src/me/haved/engine/core/FPSCounter.java
package me.haved.engine.core;

public class FPSCounter {
	private int frameCount;
	private long startTimeMillis = System.currentTimeMillis(); 
	public void nextFrame() {
		frameCount++;
		long diff = System.currentTimeMillis() - startTimeMillis;
		if(diff>=1000) {
			System.out.println("FPS:"+(frameCount*1000f/diff));
			frameCount = 0;
			startTimeMillis = System.currentTimeMillis();
		}
	}
}

File: oldStuff/cppCode/SmithGame/src/FPSCounter.cpp
#include "FPSCounter.h"

#include <iostream>

FPSCounter::FPSCounter() : m_prevPrintout(0), m_framesSincePrintout(0) {}

void FPSCounter::NextFrame() {
    m_framesSincePrintout++;
    uint32_t diff = SDL_GetTicks() - m_prevPrintout;
    if(diff>=1000) {
        std::cout << "FPS: " << (m_framesSincePrintout*1000.f/diff) << std::endl;
        m_prevPrintout=SDL_GetTicks();
        m_framesSincePrintout = 0;
    }
}
